src/Symfony/Component/Cache/Tests/Adapter/PdoDbalAdapterTest.php

```diff
@@ -43,4 +46,68 @@ public function createCachePool(int $defaultLifetime = 0): CacheItemPoolInterfac
     {
         return new PdoAdapter(DriverManager::getConnection(['driver' => 'pdo_sqlite', 'path' => self::$dbFile]), '', $defaultLifetime);
     }
+
+    public function testConfigureSchemaDecoratedDbalDriver()
+    {
+        $connection = DriverManager::getConnection(['driver' => 'pdo_sqlite', 'path' => self::$dbFile]);
+        if (interface_exists(\Doctrine\DBAL\Driver\Middleware::class, false)) {
+            $middleware = $this->createMock(\Doctrine\DBAL\Driver\Middleware::class);
+            $middleware
+                ->method('wrap')
+                ->willReturn(new DriverWrapper($connection->getDriver()));
+
+            $config = new Configuration();
+            $config->setMiddlewares([$middleware]);
+
+            $connection = DriverManager::getConnection(['driver' => 'pdo_sqlite', 'path' => self::$dbFile], $config);
+        } else {
```

Comment:
This is for `doctrine/dbal v2`, which doesn't have the middleware. Not sure if the test should cover this or something like a `SkippedTestSuiteError` should be thrown.

Reply:
I'd rather call `$this->markTestSkipped` as the feature does not exist properly for v2

Reply:
I agree. If the test is irrelevant for DBAL 2, just skip it if we‘re not testing against DBAL 3.

Reply:
> I agree. If the test is irrelevant for DBAL 2, just skip it if we‘re not testing against DBAL 3.

I don't think that this fix is irrelevant for DBAL 2, because you could still decorate/wrap the driver manually anyway, see #42022 and the originating issue https://github.com/getsentry/sentry-symfony/issues/530.

In general, I think we should change the `switch` to use `$this->conn->getDatabasePlatform()->getName()` as a reliable source of information.

Reply:
> because you could still decorate/wrap the driver manually anyway

But not as official dbal v2 functionality right? You can pass a `driverClass` to the configuration, which is then constructed without arguments, making it less usable to function as decorator. Or it'd be via Reflection changing the private property `_driver` to public at runtime. Which is in my opinion already such an edge case, that Symfony shouldn't have to test against that.